contracts/rewards_fuses/ramses/RamsesClaimFuse.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {INonfungiblePositionManagerRamses} from "../../fuses/ramses/ext/INonfungiblePositionManagerRamses.sol";
+import {PlasmaVaultLib} from "../../libraries/PlasmaVaultLib.sol";
+
+/**
+ * @title RamsesClaimFuse
+ * @dev Contract for claiming rewards for NFT positions in the Ramses system.
+ */
+contract RamsesClaimFuse {
+    using SafeERC20 for IERC20;
+
+    address public immutable VERSION;
+    /// @notice Address of the non-fungible position manager
+    address public immutable NONFUNGIBLE_POSITION_MANAGER;
+
+    /// @notice Error thrown when the lengths of tokenIds and tokenRewards arrays do not match
+    error RamsesClaimFuseTokenIdsAndTokenRewardsLengthMismatch();
+
+    /// @notice Event emitted after a reward is transferred
+    /// @param tokenId The ID of the token
+    /// @param token The address of the token
+    /// @param reward The amount of the reward
+    event RamsesClaimFuseTransferredReward(uint256 tokenId, address token, uint256 reward);
+
+    /**
+     * @dev Constructor for the contract
+     * @param nonfungiblePositionManager_ The address of the non-fungible position manager
+     */
+    constructor(address nonfungiblePositionManager_) {
+        VERSION = address(this);
+        NONFUNGIBLE_POSITION_MANAGER = nonfungiblePositionManager_;
+    }
+
+    /**
+     * @notice Function to claim rewards for multiple positions
+     * @param tokenIds Array of token IDs
+     * @param tokenRewards Array of arrays of reward token addresses
+     */
+    function claim(uint256[] memory tokenIds, address[][] memory tokenRewards) external {
```

Comment:
We are not doing this because:
- On the Ramses side, it is checked whether the tokenId belongs to PlasmaVault.
- I don't see the need; we can only claim rewards from NFTs for which we have permissions.